lib/linked_list.rb

```diff
@@ -19,57 +19,132 @@ def initialize
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
     def add_first(value)
-      raise NotImplementedError
+      @head = Node.new(value, @head)
     end
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
     def search(value)
-      raise NotImplementedError
+        if @head.nil?
+            return nil
+        end
+
+        until @head.nil?

```

Comment:
This is going to lose all subsequent elements of the linked list.

You need to have a current element to traverse the list.

```ruby
  return nil if @head.nil?

  current = @head
  until current.next.nil?
    return true if current.data == value
    current = current.next
  end
return false
```